### ISSUE TITLE
Invalid parameter not satisfying: multi

### ISSUE THREAD

**Issue by mmackh** [karelia/ConnectionKit#43]
I pulled the most recent version of ConnectionKit and when I try reading the contents of a directory now, it'll give me the following error:

```
2013-06-21 18:39:45.290 Orbitink[41117:1803] *** Assertion failure in -[CURLTransfer initWithRequest:credential:delegate:delegateQueue:multi:], /Users/imariettaone/Desktop/Obj-C/Orbitink/ConnectionKit/CurlHandle/CURLHandleSource/CURLTransfer.m:182
2013-06-21 18:39:45.291 Orbitink[41117:1803] An uncaught exception was raised
2013-06-21 18:39:45.291 Orbitink[41117:1803] Invalid parameter not satisfying: multi
2013-06-21 18:39:45.292 Orbitink[41117:1803] (
    0   CoreFoundation                      0x00007fff8e311b06 __exceptionPreprocess + 198
    1   libobjc.A.dylib                     0x00007fff8c7723f0 objc_exception_throw + 43
    2   CoreFoundation                      0x00007fff8e311948 +[NSException raise:format:arguments:] + 104
    3   Foundation                          0x00007fff8f97fc82 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 189
    4   CURLHandle                          0x000000010010e77a -[CURLTransfer initWithRequest:credential:delegate:delegateQueue:multi:] + 250
    5   Connection                          0x00000001000cfa92 -[CK2CURLBasedProtocol startWithCredential:] + 338
    6   Connection                          0x00000001000d0a70 -[CK2CURLBasedProtocol useCredential:forAuthenticationChallenge:] + 144
    7   Connection                          0x00000001000d4dc7 __81-[CK2AuthenticationChallengeTrampoline useCredential:forAuthenticationChallenge:]_block_invoke + 87
    8   libdispatch.dylib                   0x00007fff9121cf01 _dispatch_call_block_and_release + 15
    9   libdispatch.dylib                   0x00007fff912190b6 _dispatch_client_callout + 8
    10  libdispatch.dylib                   0x00007fff9121a47f _dispatch_queue_drain + 235
    11  libdispatch.dylib                   0x00007fff9121a2f1 _dispatch_queue_invoke + 52
    12  libdispatch.dylib                   0x00007fff9121a1c3 _dispatch_worker_thread2 + 249
    13  libsystem_c.dylib                   0x00007fff8f124d0b _pthread_wqthread + 404
    14  libsystem_c.dylib                   0x00007fff8f10f1d1 start_wqthread + 13
)
2013-06-21 18:39:45.292 Orbitink[41117:1803] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid parameter not satisfying: multi'
*** First throw call stack:
(
    0   CoreFoundation                      0x00007fff8e311b06 __exceptionPreprocess + 198
    1   libobjc.A.dylib                     0x00007fff8c7723f0 objc_exception_throw + 43
    2   CoreFoundation                      0x00007fff8e311948 +[NSException raise:format:arguments:] + 104
    3   Foundation                          0x00007fff8f97fc82 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 189
    4   CURLHandle                          0x000000010010e77a -[CURLTransfer initWithRequest:credential:delegate:delegateQueue:multi:] + 250
    5   Connection                          0x00000001000cfa92 -[CK2CURLBasedProtocol startWithCredential:] + 338
    6   Connection                          0x00000001000d0a70 -[CK2CURLBasedProtocol useCredential:forAuthenticationChallenge:] + 144
    7   Connection                          0x00000001000d4dc7 __81-[CK2AuthenticationChallengeTrampoline useCredential:forAuthenticationChallenge:]_block_invoke + 87
    8   libdispatch.dylib                   0x00007fff9121cf01 _dispatch_call_block_and_release + 15
    9   libdispatch.dylib                   0x00007fff912190b6 _dispatch_client_callout + 8
    10  libdispatch.dylib                   0x00007fff9121a47f _dispatch_queue_drain + 235
    11  libdispatch.dylib                   0x00007fff9121a2f1 _dispatch_queue_invoke + 52
    12  libdispatch.dylib                   0x00007fff9121a1c3 _dispatch_worker_thread2 + 249
    13  libsystem_c.dylib                   0x00007fff8f124d0b _pthread_wqthread + 404
    14  libsystem_c.dylib                   0x00007fff8f10f1d1 start_wqthread + 13
)
libc++abi.dylib: terminate called throwing an exception
(lldb) 
```

Is this something I can track down in my code or is this framework related?

**Comment by mikeabdullah** [comment]
This would be a really dumb mistake on my part where the unit tests work, but not real-world usage. Hold tight.

**Comment by mikeabdullah** [comment]
Fixed in 7f4cd30679799af1f4643ee6b8db0efbd44ba9bb. Sorry about that!

**Comment by mmackh** [comment]
Thank you Mike for the quick fix - it's working now
